pkg/k8s/apis/cilium.io/v2alpha1/bgpp_types.go

```diff
@@ -97,48 +113,47 @@ type CiliumBGPNeighbor struct {
 	// +kubebuilder:validation:Minimum=1
 	// +kubebuilder:validation:Maximum=65535
 	// +kubebuilder:default=179
-	PeerPort *int `json:"peerPort,omitempty"`
+	PeerPort *int32 `json:"peerPort,omitempty"`
 	// PeerASN is the ASN of the peer BGP router.
 	// Supports extended 32bit ASNs
 	//
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:Minimum=0
 	// +kubebuilder:validation:Maximum=4294967295
-	PeerASN int `json:"peerASN"`
+	PeerASN int64 `json:"peerASN"`
 	// EBGPMultihopTTL controls the multi-hop feature for eBGP peers.
 	// Its value defines the Time To Live (TTL) value used in BGP packets sent to the neighbor.
 	// When empty or zero, eBGP multi-hop feature is disabled. The value is ignored for iBGP peers.
 	//
 	// +kubebuilder:validation:Optional
 	// +kubebuilder:validation:Minimum=0
 	// +kubebuilder:validation:Maximum=255
-	EBGPMultihopTTL int `json:"eBGPMultihopTTL,omitempty"`
-	// ConnectRetryTime defines the initial value for the BGP ConnectRetryTimer (RFC 4271, Section 8).
-	// The default value for the ConnectRetryTime (if empty or zero) is 120 seconds.
-	// Rounded internally to the nearest whole second.
+	EBGPMultihopTTL *int32 `json:"eBGPMultihopTTL,omitempty"`
```

Comment:
Note that `EBGPMultihopTTL` is not defaulted explicitly - I thought it would not make sense to default it, as it has no meaning for iBGP peers. Does it make sense or should we default it `0`?

Reply:
> Have you tested whether creating a CiliumBGPPeeringPolicy without this field set will cause validation to fail?

Sure, it will not cause a validation error. Validation only applies if the value is set.

Reply:
Since we indicate this option is for eBGP only, maybe setting the default to 1 (and ignoring that value for iBGP internally) makes our life a bit easier and the API looks consistent?

Reply:
> maybe setting the default to 1 (and ignoring that value for iBGP internally) makes our life a bit easier and the API looks consistent?

I like that idea, it will also address the inconsistency with the GR discussed in the other thread.

Reply:
Done in https://github.com/cilium/cilium/pull/26040/commits/9958110ed9b17fc70180d6617ab2c34ef29a1b10